.github/workflows/full-checks.yml

```diff
@@ -0,0 +1,62 @@
+name: "Full checks"
+
+on:
+  pull_request:
+
+jobs:
+  full-checks:
+    name: "Full CI checks"
+
+    runs-on: ${{ matrix.operating-system }}
+
+    strategy:
+      matrix:
+        dependencies:
+          - "locked"
+        php-version:
+          - "7.4"
+        operating-system:
+          - "ubuntu-latest"
+
+    steps:
+      - name: "Checkout"
+        uses: "actions/checkout@v2"
+
+      - name: "Install PHP"
+        uses: "shivammathur/setup-php@v2"
+        with:
+          coverage: Xdebug
+          php-version: "${{ matrix.php-version }}"
+          ini-values: memory_limit=-1
+          tools: composer
+
+      - name: "Cache dependencies"
+        uses: "actions/cache@v2"
+        with:
+          path: |
+            ~/.composer/cache
+            vendor
```

Comment:
You shouldn't cache `vendor/`, see https://github.com/shivammathur/setup-php#cache-composer-dependencies
> Please do not cache vendor directory using action/cache as that will have side-effects.

There is also a/the guide how to cache correctly

Reply:
Hey. Markus. Thanks for these comments. Very much appreciated. I shall fix them shortly.